### DISCUSSION TITLE
mAP=0 on locally trained model

### DISCUSSION THREAD

**Discussion by traeplejer** [ultralytics/yolov5#10256] [Q&A]
Six months ago YOLOv5 worked locally on my pc without a problem, though it took me a while to figure out the right combination of versions. I could train and detect with costum datasets, with my RTX T600 GPU on Windows 10 Pro 64bit

First I installed Anaconda3, and in the conda promt:
conda create -n ML38 python=3.8
conda activate ML38
conda install pytorch==1.10.1 torchvision==0.11.2 torchaudio==0.10.1 cudatoolkit=11.3 -c pytorch -c conda-forge
conda install -c anaconda git
cd E:\ML
git clone https://github.com/ultralytics/yolov5
cd yolov5
pip install -r requirements.txt
pip install wandb
wandb login
(valid and train folders in ML, same as yolov5, and data.yaml in yolov5/data)

Now I have installed YOLOv5 the exact same way again, and it seems to be working, but mAP stays at 0 through all epochs, when training with the same dataset as before. Detect works fine on cars, people etc. but not on the custom object.

I have tried different versions of python and CUDAtoolkit, but only the above version is working-ish.

I don´t suppose it is because of the Zero-mAP bug that has been resolved?
https://giter.vip/ultralytics/yolov5/issues/9059

I understand that it might be an environment problem, but now I am stuck? I am now in the process of learning how to use Google colab, but I would prefer to train locally.

As a biologist I have very little experience with computer science, so please excuse my very basic approach and knowledge. I hope to use YOLOv5 for my masters thesis on wildlife monitoring with drones, so I would be very grateful for any help.

**Comment by glenn-jocher** [comment]
@traeplejer I'd recommend linux for all DL. If you're forced to use windows you can try our Docker image for a local linux environment, or Colab is super easy to start with also.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
- **Docker Image**. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

**Reply by traeplejer** [reply]
Colab really is easy! I think it just might cover my needs. If not, I´m gonna up my linux and/or Docker skills.

Thank you for the reply, and for all your work with YOLOv5.
Best regards
